instrumentation/jvm-metrics/src/main/java/com/splunk/opentelemetry/instrumentation/jvmmetrics/AllocatedMemoryMetrics.java

```diff
@@ -53,13 +52,22 @@ private static boolean hasComSunThreadMXBean() {
     }
   }
 
+  private static boolean mxBeanTypeIsCompatible() {
+    try {
+      Class<?> mxBeanClass = Class.forName("com.sun.management.ThreadMXBean");
```

Comment:
use `Class.forName("com.sun.management.ThreadMXBean", false, AllocatedMemoryMetrics.class.getClassLoader());` like in `hasComSunThreadMXBean()` or load the class in separate method and keep it in a static field or pass it to these methods